File: src/main/java/com/example/chatService/oauth/OAuthApi.java
package com.example.chatService.oauth;

import com.example.chatService.dto.UserDto;
import com.example.chatService.entity.User;
import com.example.chatService.repository.UserRepository;
import com.example.chatService.security.UserPrincipal;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api")
@RequiredArgsConstructor
@Slf4j
public class OAuthApi {

    private final UserRepository userRepository;
    @PostMapping("/user/oauth/nickname")
    public ResponseEntity<UserDto> setNickname(
            @RequestBody Map<String, String> body,
            Authentication authentication
    ) {
        String nickname = body.get("nickname");

        UserPrincipal principal = (UserPrincipal) authentication.getPrincipal();
        User user = userRepository.findById(principal.getId()).orElseThrow();

        user.setUsername(nickname);
        user.setNicknameCompleted(true);

        userRepository.save(user);

        return ResponseEntity.ok(UserDto.from(user));
    }

    @GetMapping("/me")
    public ResponseEntity<UserDto> me(Authentication authentication) {

        UserPrincipal principal = (UserPrincipal) authentication.getPrincipal();

        User user = userRepository.findById(principal.getId())
                .orElseThrow();

        return ResponseEntity.ok(UserDto.from(user));
    }
}

File: src/main/java/com/example/chatService/oauth/OAuthApi.kt
package com.example.chatService.oauth

import com.example.chatService.dto.OAuthNicknameRequest
import com.example.chatService.dto.UserDto
import com.example.chatService.repository.UserRepository
import com.example.chatService.security.UserPrincipal
import org.springframework.http.ResponseEntity
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class OAuthApi (
        private val userRepository: UserRepository
){
    @PostMapping("/user/oauth/nickname")
    fun setNickname(@RequestBody request : OAuthNicknameRequest,
                    authentication : Authentication) : ResponseEntity<UserDto> {
        val nickname = request.nickname

        val principal = authentication.principal as UserPrincipal
        val user = userRepository.findById(principal.id).orElseThrow()

        user.usernameValue = nickname
        user.nicknameCompleted = true

        userRepository.save(user)

        return ResponseEntity.ok(UserDto.from(user))
    }

    @GetMapping("/me")
    fun me(authentication: Authentication) : ResponseEntity<UserDto> {
        val principal = authentication.principal as UserPrincipal

        val user = userRepository.findById(principal.id).orElseThrow()

        return ResponseEntity.ok(UserDto.from(user))
    }
}